src/main/src/main/java/org/geoserver/catalog/CoverageViewReader.java

```diff
@@ -388,12 +420,106 @@ public GridCoverage2D read(GeneralParameterValue[] parameters) throws IllegalArg
         return result;
     }
 
+    private GridCoverage2D readUsingJiffle(ViewInputs viewInputs) throws IOException {
+        // TODO: Optimize this to introduce a bandSelect and bands remap when dealing
+        // with coverages with many bands.
+        HashMap<String, GridCoverage2D> inputCoverages = viewInputs.getInputCoverages();
```

Comment:
Since we can know the input names by parsing the script, reading only the necessary inputs should be quick, and will improve performance significantly for remote sources, or case where immediate loading is configured.

Reply:
That's already how it works. When saving the CoverageView, the Jiffle parser takes note of the input coverages being used by the script. This information is attached to the coverageView and the coverageReader will only read the needed coverages.